app/src/main/java/com/example/triptracker/view/profile/UserProfileOverview.kt

```diff
@@ -72,176 +67,157 @@ import com.example.triptracker.viewmodel.loggedUser
  */
 @Composable
 fun UserProfileOverview(
-    userProfileViewModel: UserProfileViewModel = viewModel(),
-    homeViewModel: HomeViewModel = viewModel(),
-    navigation: Navigation
+    navigation: Navigation,
+    profile: MutableUserProfile,
+    homeViewModel: HomeViewModel = viewModel()
 ) {
-  val userMail: String = loggedUser.email ?: ""
-  var readyToDisplay by remember { mutableStateOf(false) }
-  var profile by remember { mutableStateOf(UserProfile("")) }
   val myTripsList = homeViewModel.filteredItineraryList
   var myTripsCount = 0
   myTripsList.observeForever(Observer { list -> myTripsCount = list.size })
 
-  userProfileViewModel.getUserProfile(userMail) { fetch ->
-    if (fetch != null) {
-      profile = fetch
-      readyToDisplay = true
-    }
+  homeViewModel.setSearchFilter(FilterType.USERNAME)
+  homeViewModel.setSearchQuery(
+      profile.userProfile.value.username) // Filters the list of trips on user that created it
+  var sizeUsername = 24.sp
+  if (profile.userProfile.value.username.length > 15) {

```

Comment:
please put this value (15) in a constant variable

Reply:
not mine too